tools/src/tools.ml

```diff
@@ -469,12 +483,13 @@ let extractDocs ~entryPointFile ~debug =
                             (extractDocsForModule
                                ~modulePath:(interface.name :: modulePath)
                                interface))
-                     | Module {type_ = Constraint (Structure m, Ident _)} ->
-                       (* module M: T = {  }. Print M *)
-                       Some
-                         (Module
-                            (extractDocsForModule
-                               ~modulePath:(m.name :: modulePath) m))
+                     | Module {type_ = Constraint (Structure m, Ident p)} ->
+                       (* module M: T = { <impl> }. Print M *)
+                       let docs =
+                         extractDocsForModule ~modulePath:(m.name :: modulePath)
+                           m
+                       in
+                       Some (Module {docs with moduletype = Some (Path.name p)})
```

Comment:
Is this the same string, which is used for the id of the module type, when generating json for the module type?

For further processing it would make sense, to use the id. If so, the record field should probably renamed as well (`moduleTypeId`).

Reply:
Renamed field to `moduletypeid` bc all fields are in lowercase.

https://github.com/rescript-lang/rescript-vscode/pull/1019/commits/a67efa57452f825da45bb7d8fd39a27895b17290 added full id `DocExtractionRes.Example`